tls/s2n_config.c

```diff
@@ -687,6 +687,9 @@ int s2n_config_set_session_tickets_onoff(struct s2n_config *config, uint8_t enab
     }
 
     config->use_tickets = enabled;
+    if (config->initial_tickets_to_send == 0) {
```

Comment:
I mean, this is a TLS1.3 field. What will happen if we set  config->initial_tickets_to_send = 1 in a TLS1.2 connection? Probably nothing tbh, but it is a little strange. Curious what your thoughts are on this.

Reply:
We would already be doing that in some cases. What if we fall back to TLS1.2 using a config configured to use tickets?

Reply:
Mind adding a not here not to call `s2n_config_set_initial_ticket_count` to avoid infinite recursion?